### PR TITLE
Mass update power levels

### PR DESCRIPTION
**Pull request by Twi1ightSparkle**
Promote a user in a bunch of rooms

### CODE REVIEW

File: mass_update_pl.py [hunk 1]
```diff
@@ -0,0 +1,238 @@
...
+            method="GET",
+            url=target,
+            headers=headers
+        )
+
+
+        # If error on GET, log it and move on to next room
+        if not power_levels.status_code == 200:
+            try:
+                content = str(power_levels.json()).replace("\n", "").replace(";", ":").replace("\"", "'")
```
Comment (Half-Shot) [inline mass_update_pl.py:131]: What is this substitution trying to do?

File: mass_update_pl.py [hunk 2]
```diff
@@ -0,0 +1,238 @@
...
+            )
+            continue
+
+
+        # Get the power_levels state event from requests data
+        power_levels = power_levels.json()
+
+
+        # Get current PL for promote_user or default to 0 if it does not have any in the room
+        try:
```
Comment (Half-Shot) [inline mass_update_pl.py:153]: You can shorten this to `power_levels.get("users", {}).get(promote_user, 0)`, which would also be safer as `users` may not exist.

Comment (Half-Shot) [reply mass_update_pl.py:153]: If the user doesn't have a custom PL, the default value is `power_levels.get("users_default", 0)`. It may not be 0

File: mass_update_pl.py [hunk 3]
```diff
@@ -0,0 +1,238 @@
...
+            old_pl = power_levels["users"][promote_user]
+        except KeyError:
+            old_pl = 0
+
+
+        # Get PL for admin_user and append promote_user to state event
+        admin_current_pl = power_levels["users"][admin_user]
+        new_pl = admin_current_pl - minus_pl
+        power_levels["users"][promote_user] = new_pl
```
Comment (Half-Shot) [inline mass_update_pl.py:162]: users may not exist here

**Review by Half-Shot — COMMENTED** [review]
Thoughts

File: mass_update_pl.py [hunk 4]
```diff
@@ -0,0 +1,238 @@
...
+# PostgreSQL query to get rooms to work with. Must return room_alias in column 0 and room_id in column 1
+rooms_query = """
+    SELECT
+        room_aliases.room_alias,
+        rooms.room_id
+    FROM
+        rooms
+        LEFT JOIN room_aliases ON rooms.room_id = room_aliases.room_id
+    WHERE
+        rooms.room_id = '!lkFcylzZFTWNDWlcjs:example.com'
```
[the remark this answers is not in the document]
Comment (Twi1ightSparkle) [reply mass_update_pl.py:33]: Changed example to an where rooms.room_in IN (list,of,rooms)

File: mass_update_pl.py (hunk 1 above)
Comment (Twi1ightSparkle) [reply mass_update_pl.py:131]: Removing line line breaks, semicolons and double quotes as these three breaks the csv log line

File: mass_update_pl.py (hunk 2 above)
Comment (Twi1ightSparkle) [reply mass_update_pl.py]: Added to code

File: mass_update_pl.py (hunk 3 above)
Comment (Twi1ightSparkle) [reply mass_update_pl.py:202]: Added a thing to not die in a fire if this happens

**Review by Half-Shot — APPROVED** [review]
